airgun/entities/hardware_model.py

```diff
@@ -37,13 +37,22 @@ def update(self, entity_name, values):
         view.flash.assert_no_error()
         view.flash.dismiss()
 
-    def delete(self, entity_name):
-        """Delete hardware model"""
+    def delete(self, entity_name, cancel_dialog=False):
+        """Delete hardware model
+
+        cancel_dialog - dialog needs to be cancelled, if action expects an error
+        """
         view = self.navigate_to(self, 'All')
         view.search(entity_name)
-        view.table.row(name=entity_name)['Actions'].widget.click(handle_alert=True)
-        view.flash.assert_no_error()
-        view.flash.dismiss()
+        view.table.row(name=entity_name)['Actions'].widget.click()
+        if cancel_dialog:
```

Comment:
https://github.com/SatelliteQE/robottelo/pull/9028/files#r722256318

related discussion going on in the robottelo PR